scripts/clean_validate.py

```diff
@@ -0,0 +1,95 @@
+
+def main(raw_data, name_file, write_to):
+    """Clean raw data and validate it."""
+    # Extract column names from .names file
+    colnames = extract_column_name(name_file)
+
+    # Read raw data
+    imported_data = read_raw_data(raw_data, colnames)
+
+    # Removing id column and relabel diagnosis column
+    cleaned_data = clean_data(imported_data)
+
+    # Create Great Expectation batch object for data validation
+    batch = create_data_batch(cleaned_data)
+
+    # Validate cleaned data
+    # Validates that the dataframe contains specific columns.
+    exp_column_exsist(batch, colnames)
+
+    # Validates that the distinct values in specified columns of a dataset contain the expected values.
+    col_set = {
+        'diagnosis': ['Malignant', 'Benign']
+    }
+    exp_value_set_in_col(batch, col_set)
+
+    # Validates that the type of values of specified column is correct.
+    col_type = {'diagnosis': 'string'}
+    for key in colnames[2:]:
+        col_type[key] = 'number'
+    exp_type_of_col_values(batch, col_type)
+
+    # Validates that the numbers of nulls of specific columns are within the tolerable limit.
+    col_percent = {'diagnosis': 0}
+    for key in colnames[2:]:
+        col_type[key] = 0.1
+    # For numerical columns, no more than 10% of values are nulls
+    exp_col_not_null(batch, col_percent)
+
+    # Validates that the numeric values of specified columns are within specified range.
+    col_range = {
+            'mean_radius': [6,30,False,False],
+            'mean_texture': [9,40,False,False],
+            'mean_perimeter': [40,200,False,False],
+            'mean_area': [140,2510,False,False],
+            'mean_smoothness': [0,1,False,False],
+            'mean_compactness': [0,1,False,False],
+            'mean_concavity': [0,1,False,False],
+            'mean_concave': [0,1,False,False],
+            'mean_symmetry': [0,1,False,False],
+            'mean_fractal': [0,1,False,False],
+            'se_radius': [0,3,False,False],
+            'se_texture': [0,5,False,False],
+            'se_perimeter': [0,22,False,False],
+            'se_area': [6,550,False,False],
+            'se_smoothness': [0,1,False,False],
+            'se_compactness': [0,1,False,False],
+            'se_concavity': [0,1,False,False],
+            'se_concave': [0,1,False,False],
+            'se_symmetry': [0,1,False,False],
+            'se_fractal': [0,1,False,False],
+            'max_radius': [7,40,False,False],
+            'max_texture': [12,50,False,False],
+            'max_perimeter': [50,260,False,False],
+            'max_area': [180,4300,False,False],
+            'max_smoothness': [0,1,False,False],
+            'max_compactness': [0,2,False,False],
+            'max_concavity': [0,2,False,False],
+            'max_concave': [0,1,False,False],
+            'max_symmetry': [0,1,False,False],
+            'max_fractal': [0,1,False,False]
+    }
```

Comment:
When I see something this long, and with values like this hard-coded into a script I start to think how we can do this better... One thing is that the range of possible values is similar across most measures, whether it's the mean or max (which makes sense). And even for some se's.... Another is the repeat of all the Falses. Finally, if we wanted to update these values, editing the script feels wrong... I feel like a config file or a data file might be best?

Reply:
Do the other expectations have similar large amounts of data or is it just this one?